### PR TITLE
linux: unpoison system call results for MSAN

### PR DESCRIPTION
**Pull request by bnoordhuis**

### CODE REVIEW

**Comment by bnoordhuis** [comment]
> If we want to enable MSAN in CI there are some tests failing.

I'm looking into that but it needs some more work. On my machine (x84_64 linux w/ clang 14.0.0) it complains about stat/fstat/lstat calls, among other things. Not intractable, I'll push more work tomorrow.

**Comment by bnoordhuis** [comment]
Yes, it's ubuntu's clang. I think I've got everything now. I added a MSAN CI bot, let's see how it fares.

**Comment by bnoordhuis** [comment]
Crap...

> cc: error: unrecognized argument to ‘-fsanitize=’ option: ‘memory’

**Comment by bnoordhuis** [comment]
Tweaked and fixed, PTAL!

**Comment by gbaraldi** [comment]
If you find it useful we could enable origin tracking in CI too, it does make debugging slightly easier while increasing the runtime a bit.

**Review by gbaraldi — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,10 +36,12 @@ if(QEMU)
   add_definitions(-D__QEMU__=1)
 endif()
 
+# Note: these are mutually exclusive.
 option(ASAN "Enable AddressSanitizer (ASan)" OFF)
+option(MSAN "Enable MemorySanitizer (MSan)" OFF)
 option(TSAN "Enable ThreadSanitizer (TSan)" OFF)
 
-if((ASAN OR TSAN) AND NOT (CMAKE_C_COMPILER_ID MATCHES "AppleClang|GNU|Clang"))
+if((ASAN OR MSAN OR TSAN) AND NOT (CMAKE_C_COMPILER_ID MATCHES "AppleClang|GNU|Clang"))
```
[the remark this answers is not in the document]
Comment (bnoordhuis) [reply CMakeLists.txt]: Yep, fixed.

**Pull request merged**